pkg/sdk/streams_def.go

```diff
@@ -0,0 +1,201 @@
+package sdk
+
+import g "github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk/poc/generator"
+
+//go:generate go run ./poc/main.go
+
+var (
+	onStreamDef = g.QueryStruct("OnStream").
+			OptionalSQL("AT").
+			OptionalSQL("BEFORE").
+			QueryStructField(
+			"Statement",
+			g.QueryStruct("OnStreamStatement").
+				OptionalTextAssignment("TIMESTAMP", g.ParameterOptions().ArrowEquals()).
+				OptionalTextAssignment("OFFSET", g.ParameterOptions().ArrowEquals()).
+				OptionalTextAssignment("STATEMENT", g.ParameterOptions().ArrowEquals()).
+				OptionalTextAssignment("STREAM", g.ParameterOptions().ArrowEquals().SingleQuotes()).
+				WithValidation(g.ExactlyOneValueSet, "Timestamp", "Offset", "Statement", "Stream"),
+			g.ListOptions().Parentheses(),
+		).
+		WithValidation(g.ExactlyOneValueSet, "At", "Before")
+
+	showStreamDbRowDef = g.DbStruct("showStreamsDbRow").
+				Field("created_on", "time.Time").
+				Field("name", "string").
+				Field("database_name", "string").
+				Field("schema_name", "string").
+				Field("tableOn", "sql.NullString").
+				Field("owner", "sql.NullString").
+				Field("comment", "sql.NullString").
+				Field("table_name", "sql.NullString").
+				Field("source_type", "sql.NullString").
+				Field("base_tables", "sql.NullString").
+				Field("type", "sql.NullString").
+				Field("stale", "sql.NullString").
+				Field("mode", "sql.NullString").
+				Field("stale_after", "sql.NullTime").
+				Field("invalid_reason", "sql.NullString").
+				Field("owner_role_type", "sql.NullString")
```

Comment:
I can't see that in the documentation. If it's there, please mention it on our #doc-discuss channel.

Reply:
Done